src/components/MarkdownItRender.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { createBrowserHistory } from 'history';
+import MarkdownIt from 'markdown-it';
+import MarkdownItMusic from 'markdown-it-music'
+import queryString from 'query-string'
+
+const history = createBrowserHistory();
+
+class MarkdownMusic extends React.Component {
+  constructor(props) {
+    super(props);
+
+    let musicOpts = {
+      transpose: this.props.transpose
+    };
+
+    this.md = MarkdownIt()
+      .use(MarkdownItMusic, musicOpts);
+  }
+
+  render() {
+    this.md.setTranspose(this.props.transpose);
+    return (
+      <span dangerouslySetInnerHTML={{ __html: this.md.render(this.props.source) }} />
+    );
+  }
+}
+
+class Markdown extends React.Component {
```

Comment:
We need a better name for this component. I'm not sure what it is, but here's what this component does:

- Retrieves the markdown file from a GitHub repo
- Provides control of the transpose
- Decodes the Markdown and renders it with MarkdownMusic

Right now, I think that this component is probably trying to do too many things. Let's make a TODO to decouple retrieval of source markdown and controlling arguments to MarkdownMusic.

Reply:
Agree, I looked at this for a little bit and it was totally initially vague and unintuitive to me.